d2l-gestures-swipe.html

```diff
@@ -0,0 +1,105 @@
+<script>
+(function() {
+	'use strict';
+
+	var Swipe = {
+
+		_maxTime: 2000,
+
+		_handleTouchStart: function(e) {
+
+			var node = this;
+
+			var tracking = {
+				start: {
+					time: new Date().getTime(),
+					x: e.touches[0].clientX,
+					y: e.touches[0].clientY
+				}
+			};
+
+			var reset = function() {
+				tracking = null;
+				node.removeEventListener('touchend', handleTouchEnd);
+				node.removeEventListener('touchermove', handleTouchMove);
+				node.removeEventListener('touchcancel', handleTouchCancel);
+			};
+
+			var handleTouchCancel = function() {
+				reset();
+				return;
+			};
+
+			var handleTouchEnd = function() {
+
+				if (!tracking || !tracking.end) {
+					return;
+				}
+
+				var elapsedTime = new Date().getTime() - tracking.start.time;
+				if (elapsedTime > window.D2L.Gestures.Swipe._maxTime) {
+					reset();
+					return;
+				}
+
+				var distanceX = tracking.end.x - tracking.start.x;
+				var distanceY = tracking.end.y - tracking.start.y;
+
+				var theta = Math.atan(Math.abs(distanceX) / Math.abs(distanceY));
```

Comment:
Calculate the direction (in degrees, converted from radians) of the swipe, which provides for more accurate determination of swipe than simply checking one of x or y distance.